tests/datatypes/test_list.py

```diff
@@ -139,6 +139,146 @@ def test_slice(self):
+
+
+    def test_list_list_comparisons(self):
+
+
+        # `this` (left list) is empty.
+
+        self.assertCodeExecution("""
+        print('>>> x = []')
+        x = []
+        print('>>> y = [1,2,3]')
+        y = [1,2,3]
+        print('>>> x > y')
+        x>y
+        print('>>> x >= y')
+        x >= y
+        print('>>> x < y')
+        x < y
+        print('>>> x <= y')
+        x <= y
+        """)
+
+        # `other` (right list) is empty
+
+        self.assertCodeExecution("""
+        print('>>> x = [1,2,3]')
+        x = [1,2,3]
+        print('>>> y = []')
+        y = []
+        print('>>> x > y')
+        x>y
+        print('>>> x >= y')
+        x >= y
+        print('>>> x < y')
+        x < y
+        print('>>> x <= y')
+        x <= y
+        """)
+
+        # both lists are empty
+        self.assertCodeExecution("""
+        print('>>> x = []')
+        x = []
+        print('>>> y = []')
+        y = []
+        print('>>> x > y')
+        x>y
+        print('>>> x >= y')
+        x >= y
+        print('>>> x < y')
+        x < y
+        print('>>> x <= y')
+        x <= y
+        """)
+
+        # `this` (left list) is shorter
+
+        self.assertCodeExecution("""
+        print('>>> x = [1,2]')
+        x = [1,2]
+        print('>>> y = [1,2,3]')
+        y = [1,2,3]
+        print('>>> x > y')
+        x>y
+        print('>>> x >= y')
+        x >= y
+        print('>>> x < y')
+        x < y
+        print('>>> x <= y')
+        x <= y
+        """)
+
+        # `other` (right list) is shorter
+
+        self.assertCodeExecution("""
+        print('>>> x = [1,2,3]')
+        x = [1,2,3]
+        print('>>> y = [1,2]')
+        y = [1,2]
+        print('>>> x > y')
+        x>y
+        print('>>> x >= y')
+        x >= y
+        print('>>> x < y')
+        x < y
+        print('>>> x <= y')
+        x <= y
+        """)
+
+
+        # comparable items aren't equal
+
+        self.assertCodeExecution("""
+        print('>>> x = [1,2]')
+        x = [1,2]
+        print('>>> y = [1,3]')
+        y = [1,3]
+        print('>>> x > y')
+        x>y
+        print('>>> x >= y')
+        x >= y
+        print('>>> x < y')
+        x < y
+        print('>>> x <= y')
+        x <= y
+        """)
+
+        self.assertCodeExecution("""
+        print('>>> x = [1,3]')
+        x = [1,3]
+        print('>>> y = [1,2]')
+        y = [1,2]
+        print('>>> x > y')
+        x>y
+        print('>>> x >= y')
+        x >= y
+        print('>>> x < y')
+        x < y
+        print('>>> x <= y')
+        x <= y
+        """)
+
+        # all items are equal
+
+        self.assertCodeExecution("""
+         print('>>> x = [1,2,3]')
+         x = [1,2,3]
+         print('>>> y = [1,2,3]')
+         y = [1,2,3]
+         print('>>> x > y')
+         x>y
+         print('>>> x >= y')
+         x >= y
+         print('>>> x < y')
+         x < y
+         print('>>> x <= y')
+         x <= y
+         """)
+
```

Comment:
The set of tests you have here looks good; however, you could template this for clarity. Each test follows a very repeatable pattern; ideally, that pattern would be factored out into something like:
```
    self.assertOrdering([1,2,3], [1,2,3], gt=False, gte=True, lt=False, lte=True)
```

Also - it's not enough to just do the comparison, you need to print the *result* of the comparison, because it's the output that is used as the basis for the test success/failure. so, lines 278/279 should effectively be:
```
    print('>>> x <= y')
    print(x <= y)
```

This gives you the debug output to know what the next line represents, and a printed version of that output.